strawberry/federation.py

```diff
@@ -115,8 +116,18 @@ def entities_resolver(self, root, info, representations):
             type = self.schema_converter.type_map[type_name]
 
             definition = cast(TypeDefinition, type.definition)
+            resolve_reference = definition.origin.resolve_reference
 
-            results.append(definition.origin.resolve_reference(**representation))
+            func_args = get_func_args(resolve_reference)
+            kwargs = representation
+
+            if "info" in func_args:
+                kwargs["info"] = info
+
+            if "root" in func_args:
+                kwargs["root"] = root
```

Comment:
is this always the value of query/mutation?

Reply:
I don't know how `root` works to be honest, but in the test it's what I send to `root_value`, and in the codebase where I am using this is None

Reply:
I'd remove it then, at least for now :)

Reply:
root removed

Reply:
root is the value of the parent of the field :)

so:

```python
@strawberry.type
class Something:
   x: int
   @strawberry.field
   def computed_x(root): # root is the same as self
       return root.x + 2
```